internal/cache/redis.go

```diff
@@ -140,3 +149,70 @@ func (r *Cache) IncrementTestInvoiceCounter(ctx goContext.Context, userId int) (
 
 	return int(counter), nil
 }
+
+func draftsKey(userId int) string {
+	return fmt.Sprintf("user:%d:drafts", userId)
+}
+
+func (r *Cache) SaveDraft(ctx goContext.Context, draftId, name string) error {
+	err := r.client.HSet(ctx, draftsKey(context.GetUserId(ctx)), draftId, name).Err()
+	if err != nil {
+		context.GetLogger(ctx).WithFields(log.Fields{
+			"draftId": draftId,
+			"userId":  context.GetUserId(ctx),
+			"error":   err,
+		}).Error("redis.saveDraft.failed")
+		return err
+	}
+	return nil
+}
+
+func (r *Cache) DeleteDraft(ctx goContext.Context, draftId string) error {
+	res, err := r.client.HDel(ctx, draftsKey(context.GetUserId(ctx)), draftId).Result()
+	if err != nil {
+		context.GetLogger(ctx).WithFields(log.Fields{
+			"draftId": draftId,
+			"userId":  context.GetUserId(ctx),
+			"error":   err,
+		}).Error("redis.deleteDraft.failed")
+		return err
+	} else if res != 1 {
+		context.GetLogger(ctx).WithFields(log.Fields{
+			"draftId": draftId,
+			"userId":  context.GetUserId(ctx),
+		}).Debug("redis.deleteDraft.notFound")
+		return &NotFoundError{"Draft", draftId}
+	}
+
+	return nil
+}
+
+func (r *Cache) GetDrafts(ctx goContext.Context) (map[string]string, error) {
+	ids, err := r.client.HGetAll(ctx, draftsKey(context.GetUserId(ctx))).Result()
+	if err != nil {
+		context.GetLogger(ctx).WithField("userId", context.GetUserId(ctx)).Error("redis.getDrafts.failed")
+		return nil, err
+	}
+
+	return ids, nil
+}
+
+func (r *Cache) GetDraft(ctx goContext.Context, draftId string) (string, error) {
+	draftName, err := r.client.HGet(ctx, draftsKey(context.GetUserId(ctx)), draftId).Result()
+	if err != nil {
+		context.GetLogger(ctx).WithFields(log.Fields{
+			"draftId": draftId,
+			"userId":  context.GetUserId(ctx),
+			"error":   err,
+		}).Error("redis.getDraft.failed")
+		return "", err
+	} else if draftName == "" {
```

Comment:
if non-existent key returns empty string, we should forbid empty name in POST endpoint